### PR TITLE
Add Get Random Paste Button

### PR DESCRIPTION
**Pull request by golubs**
Der neue Knopf "Get Random Pasta" liefert einen zufälligen Paste aus der Datenbank.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,13 +4,24 @@ require("dotenv").config();
...
 app.use(express.json());
 
+app.get("/random", async (request, response) => {
+  try {
+    const randomId = await getRandomId();
+    const paste = await getPaste(randomId);
+    console.log(paste);
+    return response.json(paste);
+  } catch (error) {
+    console.error(error);
```
Comment (aruppert) [inline index.js:24]: no logs :)

File: lib/pastes.js [hunk 2]
```diff
@@ -22,5 +22,15 @@ async function getPaste(pasteId) {
   return paste;
 }
 
+async function getRandomId() {
+  const pastesCollection = await getCollection(collectionName);
+  const cursor = await pastesCollection.aggregate([{ $sample: { size: 1 } }]);
+  const randomPaste = await cursor.toArray();
```
Comment (lmachens) [inline lib/pastes.js]: `randomPaste` is an array. I would rename it

File: lib/pastes.js [hunk 3]
```diff
@@ -22,5 +22,15 @@ async function getPaste(pasteId) {
...
+  const { _id } = randomPaste[0];
```
Comment (lmachens) [inline lib/pastes.js]: I would return a single paste and add a "not found" case

File: package.json [hunk 4]
```diff
@@ -25,7 +25,7 @@
   "dependencies": {
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
-    "mongodb": "^3.3.2"
+    "mongodb": "^3.3.5"
```
Comment (lmachens) [inline package.json]: Why update?

File: index.js [hunk 5]
```diff
@@ -4,13 +4,24 @@ require("dotenv").config();
...
 const { initDatabase } = require("./lib/database");
-const { getPaste, setPaste } = require("./lib/pastes");
+const { getPaste, setPaste, getRandomId } = require("./lib/pastes");
 
 const app = express();
 
 // Parse application/json for all request
 app.use(express.json());
 
+app.get("/random", async (request, response) => {
```
Comment (lmachens) [inline index.js]: I would rename it to `/api/pastes/random`

File: index.js [hunk 6]
```diff
@@ -4,13 +4,24 @@ require("dotenv").config();
...
+  try {
+    const randomId = await getRandomId();
+    const paste = await getPaste(randomId);
```
Comment (lmachens) [inline index.js]: You already have access to the paste in the previous function. `getPaste` is not necessary

File: client/src/pages/Home.js [hunk 7]
```diff
@@ -17,6 +20,13 @@ export default function Home() {
   const [pasteValue, setPasteValue] = React.useState("");
   const [{ pasteId, error, loading }, doPost] = usePostPaste();
 
+  const history = useHistory();
+
+  async function handleClick() {
+    const randomId = await getRandomPasteId();
```
Comment (lmachens) [inline client/src/pages/Home.js]: `randomId` is an object

File: client/src/components/RandomButton.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import React from "react";
+import Button from "./Button";
+
+export default function RandomButton(props) {
```
Comment (lmachens) [inline client/src/components/RandomButton.js:4]: I wouldn't create a new component for that

**Review by lmachens — CHANGES_REQUESTED** [review]

**Pull request merged**